apluslms_roman/backends/__init__.py

```diff
@@ -26,21 +27,32 @@ class BuildStep:
     cmd: If not None, docker command
     mnt: If not None, course data is mounted to this path in RW mode
     env: If not None, dict that is given as environment for the image
+    ref: Name/index of the step
     """
-    __slots__ = ('img', 'cmd', 'mnt', 'env')
+    __slots__ = ('img', 'cmd', 'mnt', 'env', 'name', 'ref')
 
     @classmethod
-    def from_config(cls, data):
-        if isinstance(data, dict):
+    def from_config(cls, index, data):
+        if isinstance(data, Mapping):
             if 'img' not in data:
                 raise RuntimeError("Missing image name (img) in step configuration: {}".format(data))
             img = clean_image_name(data['img'])
-            return cls(img, data.get('cmd'), data.get('mnt'), data.get('env'))
+            return cls(
+                img,
```

Comment:
`clean_image_name(data['img'])` can be moved here as `cls(...)` is now on multiple lines,